torch/distributed/_tensor/device_mesh.py

```diff
@@ -169,15 +170,17 @@ def _init_process_groups(self):
                     f"has mesh {other_mesh}!"
                 )
 
-        # groups created by dimension, each dimension should have exact
-        # one valid process group per rank
-        dim_groups: List[ProcessGroup] = []
+        # group ranks/tags associated with each mesh dimension, each mesh dimension should
+        # have one sub-group per rank
+        dim_group_ranks: List[List[int]] = []
+        dim_group_tags: List[str] = []
 
         if self.mesh.ndim == 1 and len(unique_mesh_values) == get_world_size():
             # if the mesh is the same as world_pg, we just append the default
             # pg to the first dim groups, as new_group cannot have the exact
             # same ranks as world
-            dim_groups.append(default_pg)
+            dim_group_ranks.append(list(range(get_world_size())))
+            dim_group_tags.append(_get_group_tag(_get_default_group()))
```

Comment:
suggestion:
then we can do something like:
`dim_groups.append((tag, list(range(world_size))))`